sycl/doc/extensions/experimental/sycl_ext_oneapi_fpga_mem.asciidoc

```diff
@@ -0,0 +1,506 @@
+= sycl_ext_oneapi_fpga_mem
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (C) 2023-2023 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+
+== Contact
+
+To report problems with this extension, please open a new issue at:
+
+https://github.com/intel/llvm/issues
+
+
+== Dependencies
+
+This extension is written against the SYCL 2020 revision 7 specification.  All
+references below to the "core SYCL specification" or to section numbers in the
+SYCL specification refer to that revision.
+
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../supported/sycl_ext_oneapi_accessor_properties.asciidoc[
+  sycl_ext_oneapi_accessor_properties.asciidoc]
+
+
+== Status
+Experimental
+
+This is an experimental extension specification, intended to provide early
+access to features and gather community feedback.  Interfaces defined in this
+specification are implemented in {dpcpp}, but they are not finalized and may
+change incompatibly in future versions of {dpcpp} without prior notice.
+*Shipping software products should not rely on APIs defined in this
+specification.*
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for FPGA devices. 
+Attempting to use this extension in
+kernels that run on other devices or backends may result in undefined
+behavior. Be aware that the compiler is not able to issue a diagnostic to
+warn you if this happens.
```

Comment:
This whole section should probably be deleted given that the extension isn't supported yet and when it is supported it will work on all backends, with the annotations ignored on non-FPGA targets.